File: paper/src/main/java/gg/grounds/GroundsPluginServerDiscovery.java
package gg.grounds;

import gg.grounds.paper.PaperDiscoveryConfig;
import gg.grounds.paper.PaperDiscoveryPublisher;
import org.bukkit.plugin.java.JavaPlugin;

public final class GroundsPluginServerDiscovery extends JavaPlugin {
    private PaperDiscoveryPublisher publisher;

    @Override
    public void onEnable() {
        PaperDiscoveryConfig config = PaperDiscoveryConfig.load(getLogger());

        publisher = new PaperDiscoveryPublisher(this, config);
        publisher.start();

        getLogger().info("Paper server discovery initialized");
    }

    @Override
    public void onDisable() {
        if (publisher != null) {
            publisher.stop();
        }
    }
}

File: paper/src/main/kotlin/gg/grounds/GroundsPluginServerDiscovery.kt
package gg.grounds

import gg.grounds.discovery.PaperDiscoveryConfig
import gg.grounds.discovery.PaperDiscoveryPublisher
import org.bukkit.plugin.java.JavaPlugin

class GroundsPluginServerDiscovery : JavaPlugin() {
    private var publisher: PaperDiscoveryPublisher? = null

    override fun onEnable() {
        val config = PaperDiscoveryConfig.load()

        publisher = PaperDiscoveryPublisher(this, config)
        publisher?.start()

        logger.info("Paper server discovery initialized")
    }

    override fun onDisable() {
        publisher?.stop()
    }
}
